src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -52,46 +53,64 @@ public partial class NodeViewModel : ViewModelBase
 
         #region public members
 
+        /// <summary>
+        /// Returns NodeModel ID
+        /// </summary>
+        public Guid Id
```

Comment:
@mjkkirschner Seems correct to me, it needs to serialize the same Id as the Node it's referencing.